### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by Ryman**
Mostly minor changes, only major things are removing `AxolotlState::try_encrypt` and `Axolotl::ratchet_keys_are_equal`. 

The `Clone` impls look a lot like they could be derived but I didn't yet look into playing with the bounds in those places.

### CODE REVIEW

**Comment by mjewkes** [comment]
Hey @Ryman - thanks for the PR!
This is our first significant OS project, and we realized today that licensing is an issue we need to put some more thought into. Unfortunately, we can't accept any contribs until we figure this out. 

I will be back in touch by Wednesday.

File: src/text_secure_v3/mod.rs [hunk 1]
```diff
@@ -118,76 +124,65 @@ impl axolotl::Axolotl for TextSecureV3{
...
-                            .map(|x|{*x})
-                            .collect();
-
-        let (rk, ck) = keys_from_kdf(&master_key[..], "WhisperText".as_bytes(),&SEED_NULL);
+        local_identity_remote_handshake_dh_secret : &DHShared<Self::IdentityKey>,
+        local_handshake_remote_identity_dh_secret : &DHShared<Self::IdentityKey>,
+        local_handshake_remote_handshake_dh_secret : &DHShared<Self::IdentityKey>)
+            -> (Self::RootKey, Self::ChainKey) {
+
+        let secrets :&[&[u8]] = &[local_identity_remote_handshake_dh_secret.to_bytes(),
```
Comment (jazzz) [inline src/text_secure_v3/mod.rs]: this code is much better than what was there before, however the variable naming deviates from the spec. Would be great for master_key to be referenced for clarity

Comment (Ryman) [inline src/text_secure_v3/mod.rs]: Alright :) Do you have a link to the part of the spec you're referencing? Might be worth adding a link for future.

**Comment by Ryman** [comment]
@mjewkes Okay, no problem

**Comment by mjewkes** [comment]
At this point, we're leaning towards setting up a CLA. This project is one that is important to us, and that we intend to maintain into the future, and is unfortunately also one that borders on patent claims from time to time. Hoping to have things finalized by the end of the week.

The protocol we're referencing is the [Axolotl Ratchet](https://github.com/trevp/axolotl/wiki)

**Comment by Ryman** [comment]
@mjewkes Any update on the licensing? I think this is mostly bitrot at this point? :scream_cat:

**Pull request closed without merging**